### PR TITLE
Battle-Log: Support custom start times in youtube tags

### PR DESCRIPTION
**Pull request by mia-pi-git**
Requested, fairly easy, and useful.

### CODE REVIEW

File: src/battle-log.ts [hunk 1]
```diff
@@ -843,9 +843,15 @@ class BattleLog {
 				const videoId = /(?:\?v=|\/embed\/)([A-Za-z0-9_\-]+)/.exec(src)?.[1];
 				if (!videoId) return {tagName: 'img', attribs: ['alt', `invalid src for <youtube>`]};
 
+				const time = getAttrib('start') || /(?:\?|&)t=([0-9]+)/.exec(src)?.[1] || false;
```
Comment (Zarel) [inline src/battle-log.ts:846]:
```suggestion
				const time = getAttrib('start') || /(?:\?|&)t=([0-9a-z]+)/.exec(src)?.[1] || false;
```

We should probably support hms notation.

File: src/battle-log.ts [hunk 2]
```diff
@@ -843,9 +843,15 @@ class BattleLog {
...
+
 				return {
 					tagName: 'iframe',
-					attribs: ['width', width, 'height', height, 'src', `https://www.youtube.com/embed/${videoId}`, 'frameborder', '0', 'allow', 'accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture', 'allowfullscreen', 'allowfullscreen'],
+					attribs: [
+						'width', width, 'height', height,
+						'src', `https://www.youtube.com/embed/${videoId}${time ? `?start=${time}` : ''}`,
```
Comment (Zarel) [inline src/battle-log.ts:852]: Does `t=` not work in embeds?

Comment (mia-pi-git) [reply src/battle-log.ts:852]: What do you mean? `url?t=` unto itself, with nothing else after it?

Comment (Zarel) [reply src/battle-log.ts:852]: No, I mean, you only accept `t=` in the URL, and you only use `start=` in the embed URL. Should we all be searching for `start=` in the input URL?

Comment (Zarel) [reply src/battle-log.ts:852]: Also, are attributes sanitized in this? What happens if I try to use an attribute of `start="<i>test"`?

File: src/battle-log.ts [hunk 3]
```diff
@@ -843,9 +843,15 @@ class BattleLog {
 				const videoId = /(?:\?v=|\/embed\/)([A-Za-z0-9_\-]+)/.exec(src)?.[1];
 				if (!videoId) return {tagName: 'img', attribs: ['alt', `invalid src for <youtube>`]};
 
+				const time = /(?:\?|&)t=([0-9]+)/.exec(src)?.[1];
```
Comment (Zarel) [inline src/battle-log.ts:846]:
```suggestion
				const time = /(?:\?|&)(?:t|start)=([0-9]+)/.exec(src)?.[1];
```

**Comment by Zarel** [comment]
ty!

**Pull request merged**